### PR TITLE
[MM-16572] Migrate "User.AnalyticsGetSystemAdminCount" to Sync by default

### LINKED ISSUE #11384: Migrate "User.AnalyticsGetSystemAdminCount" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16572)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `AnalyticsGetSystemAdminCount` in the `User` store.

The expected way to implement it is, go to the `User` store implementation in the `store/sqlstore/` directory, modify the method `AnalyticsGetSystemAdminCount` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by SezalAgrawal**
#### Summary

Migrate "User.AnalyticsGetSystemAdminCount" to Sync by default
#### Ticket Link

Fixes #11384

### CODE REVIEW

File: store/sqlstore/user_store.go [hunk 1]
```diff
@@ -1429,14 +1429,12 @@ func (us SqlUserStore) AnalyticsGetInactiveUsersCount() (int64, *model.AppError)
...
 }
 
-func (us SqlUserStore) AnalyticsGetSystemAdminCount() store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if count, err := us.GetReplica().SelectInt("SELECT count(*) FROM Users WHERE Roles LIKE :Roles and DeleteAt = 0", map[string]interface{}{"Roles": "%system_admin%"}); err != nil {
-			result.Err = model.NewAppError("SqlUserStore.AnalyticsGetSystemAdminCount", "store.sql_user.analytics_get_system_admin_count.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = count
+func (us SqlUserStore) AnalyticsGetSystemAdminCount() (int64, *model.AppError) {
+		count, err := us.GetReplica().SelectInt("SELECT count(*) FROM Users WHERE Roles LIKE :Roles and DeleteAt = 0", map[string]interface{}{"Roles": "%system_admin%"})
```
Comment (jespino) [inline store/sqlstore/user_store.go]: Wrong indentation here, please use `gofmt` (probably the best way to have it is integrated directly with your editor).

Comment (SezalAgrawal) [reply store/sqlstore/user_store.go]: Got it! I have made the change

File: store/sqlstore/user_store.go [hunk 2]
```diff
@@ -1429,14 +1429,12 @@ func (us SqlUserStore) AnalyticsGetInactiveUsersCount() (int64, *model.AppError)
...
-		if count, err := us.GetReplica().SelectInt("SELECT count(*) FROM Users WHERE Roles LIKE :Roles and DeleteAt = 0", map[string]interface{}{"Roles": "%system_admin%"}); err != nil {
-			result.Err = model.NewAppError("SqlUserStore.AnalyticsGetSystemAdminCount", "store.sql_user.analytics_get_system_admin_count.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = count
-		}
-	})
+func (us SqlUserStore) AnalyticsGetSystemAdminCount() (int64, *model.AppError) {
+	count, err := us.GetReplica().SelectInt("SELECT count(*) FROM Users WHERE Roles LIKE :Roles and DeleteAt = 0", map[string]interface{}{"Roles": "%system_admin%"})
+	if err != nil {
+		return count, model.NewAppError("SqlUserStore.AnalyticsGetSystemAdminCount", "store.sql_user.analytics_get_system_admin_count.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (hanzei) [inline store/sqlstore/user_store.go]: This should return `0` as the first return value.

Comment (SezalAgrawal) [reply store/sqlstore/user_store.go]: Made the change.

**Pull request merged**